Source/Scene/Cesium3DTileset.js

```diff
@@ -1791,9 +1908,104 @@ define([
         var selectedTiles = tileset._selectedTiles;
         var length = selectedTiles.length;
         var tileVisible = tileset.tileVisible;
-        for (var i = 0; i < length; ++i) {
-            var tile = selectedTiles[i];
-            if (tile.selected) {
+
+        var tile, i;
+
+        if (tileset._hasMixedContent && frameState.context.stencilBuffer && length > 0) {
+            /**
+             * Consider 'leaf' tiles as selected tiles that have no selected descendants. They may have children,
```

Comment:
In comments and in any variable names, do not overload the term leaf.  In my thesis, I called these "dynamic leafs" - that could be OK in this context.  "Effective leaf" like you have below is perhaps better.